src/libexpr/eval.hh

```diff
@@ -582,25 +582,41 @@ struct EvalSettings : Config
     Setting<Strings> nixPath{
         this, {}, "nix-path",
         R"(
-          List of directories to be searched for `<...>` file references.
+          List of directories to be searched for file references in `<...>` search path expressions.
+          It is overridden by the [`NIX_PATH`](@docroot@/command-ref/env-common.md#env-NIX_PATH)
+          environment variable, and can be extended using the [`-I` option](@docroot@/command-ref/opt-common.md#opt-I).
 
-          If [pure evaluation](#conf-pure-eval) is disabled,
-          this is initialised using the [`NIX_PATH`](@docroot@/command-ref/env-common.md#env-NIX_PATH)
-          environment variable, or, if it is unset and [restricted evaluation](#conf-restrict-eval)
-          is disabled, a default search path including the user's and `root`'s channels.
+          When nothing is specified:
+
+          - If [pure evaluation](#conf-pure-eval) is enabled, search paths are not resolved at all.
+
+          - If [restricted evaluation](#conf-restrict-eval) is enabled, use the values in `NIX_PATH`.
+
+          - Otherwise, and if `NIX_PATH` is not set, use a fallback:
```

Comment:
Similarly this phrasing implies that the fallback is not used if restricted evaluation is enabled, but that's not true:
```
# unset NIX_PATH
# nix eval --restrict-eval --impure --expr 'builtins.nixPath'
[ { path = "/home/eelco/.nix-defexpr/channels"; prefix = ""; } { path = "/nix/var/nix/profiles/per-user/root/channels"; prefix = ""; } ]
```

Reply:
Is this still true on `master`? #7689 was supposed to fix that, and this is documenting it.